pulp_container/app/registry_api.py

```diff
@@ -609,12 +636,21 @@ def put(self, request, path, pk=None):
             except IntegrityError:
                 pass
 
-            with repository.new_version() as new_version:
-                new_version.add_content(models.Blob.objects.filter(pk=blob.pk))
-
             upload.delete()
 
-            return BlobResponse(blob, path, 201, request)
+            task = enqueue_with_reservation(
+                add_and_remove,
+                [f"upload:{pk}", repository, blob],
```

Comment:
Is it appropriate to include blob here? Or do we never lock on Content units?

Reply:
imo, we should never lock on content units. although in the 'tag' task we are locking on manifest and we should not.